Add script for extracting gene annotation table from GTF

pinin4fjords opened the pull request:
Added a simple script to produce a gene annotation table with mitochondrial genes flagged. Mitochondrial gene assignment is based on chromosome name and gene biotype.

Hopefully this will work to incorporate into a Scanpy-driven workflow, and allow proper use of e.g. mitochondrial gene expression as a QC filtering step.

Tested locally with Planemo.

pinin4fjords commented:
@pcm32 - now a generic script covering the original util function as well. Also took the opportunity to parameterise for future-proofing (e.g. other mitochondrial chromosome names).

nh3 commented on tools/util/gtf2featureAnnotation.xml:
```diff
@@ -0,0 +1,74 @@
+        <conditional name="mito">
+          <param name="mark_mito" type="boolean" checked="true" label="Flag mitochondrial features?"/>
+          <when value="true">
+            <param name="mito_chr" type="text" optional='true' value="mt,mitochondrion_genome,mito" label="Comma-separated list of possible mitochondrial chromosome names (case insensitive)"/>
```
Perhaps change the default value to something like "M,MT,chrM,chrMt"?

pinin4fjords replied:
So, it's case insensitive, and these are the chromosome names our existing annotations have for mitochondria. But it can be fairly exhaustive- I'll include m, chrm and chrMt as well.

nh3 commented on tools/util/gtf2featureAnnotation.xml:
```diff
@@ -0,0 +1,74 @@
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+       ln -s '$gtf_input' input.gtf;
+       $__tool_directory__/gtf2featureAnnotation.R --gtf-file input.gtf
```
Didn't know one could access scripts use `$__tool_directory__`, cool!

pcm32 commented:
Great work! Given that the old one was absorbed by the newer tool, I would prefer if the new tool uses the same  id, name (possibly with only id it might be enough) and increased version, so that we don't create an extra tool in the sidebar, an extra entry on the toolshed and we don't need to have two tools for backward compatibility. Thanks.

pinin4fjords commented:
@pcm32 - I've now added another function to this that I will need for training (and for doing Alevin via Galaxy for production if we ever do that). Namely, when we use this script to generate a transcript-gene mapping (needed by Alevin et al), also filter out any cDNAs from a file which are not represented in those mappings (i.e. they're missing from the GTF). Alevin dies if you don't do this synchronisation. 

I also added the possibility of 'versioning' the transcripts- often transcriptome cDNA files from Ensembl have versions on them.

I've also reverted to tool ID. Are we okay to merge?

pcm32 commented:
As discussed, we could add something like this to automatically update tool wrappers into the toolshed: https://github.com/bgruening/galaxytools/blob/master/.travis.yml#L46

I reckon is un-related, so if you want we could do it at a later PR.

pinin4fjords commented:
Great idea, but does look a little involved, so another PR in the near future is the best I think.
